plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java

```diff
@@ -732,6 +735,67 @@ public void renameMaterializedView(ConnectorSession session, SchemaTableName sou
         return listNamespaces(session);
     }
 
+    @Override
+    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName tableName)
```

Comment:
This doesn't seem like an Iceberg-catalog level method. Can we narrow this down to just the things that are dependent on the catalog implementation and move the rest up to `IcebergMetadata`?

Reply:
> This doesn't seem like an Iceberg-catalog level method.

Initially I created a `TableRedirectionHandler` abstraction which was created at the same time as the `TrinoCatalog` instance, but I dropped afterwards the idea because the redirection is tightly linked to the metastore implementation (hive/glue).

The same boilerplate code used for creating the TrinoCatalog would need to be done for redirection handling as well.

Reply:
On a second thought, it probably makes sense to take out this operation from TrinoCatalog in hindsight of JDBC / REST catalogs which will not have anymore Hive related content in them.

Reply:
I've modified the code again so that the `redirectTable` method lives in the `TrinoHiveCatalog` and `TrinoGlueCatalog`, but it is not exposed in the `TrinoCatalog` interface in order to keep the interface free from the concept of table redirection, but still be able to provide this functionality for hive & glue.